pkg/common/scheduler/statefulset/scheduler.go

```diff
@@ -160,73 +197,237 @@ func (s *StatefulSetScheduler) Schedule(vpod scheduler.VPod) ([]duckv1alpha1.Pla
 	return placements, nil
 }
 
-func (s *StatefulSetScheduler) removeReplicas(diff int32, placements []duckv1alpha1.Placement) []duckv1alpha1.Placement {
+func (s *StatefulSetScheduler) removeReplicas(diff int32, placements []duckv1alpha1.Placement, evenSpread int32) []duckv1alpha1.Placement {
 	newPlacements := make([]duckv1alpha1.Placement, 0, len(placements))
-	for i := 0; i < len(placements); i++ {
-		if diff >= placements[i].VReplicas {
-			// remove the entire placement
-			diff -= placements[i].VReplicas
-		} else {
-			newPlacements = append(newPlacements, duckv1alpha1.Placement{
-				PodName:   placements[i].PodName,
-				VReplicas: placements[i].VReplicas - diff,
-			})
-			diff = 0
+	logger := s.logger.With("removeReplicas()", zap.Int32("schedule with even spread:", evenSpread))
```

Comment:
I think you want `s.logger.Named("remove replicas").With(zap.Int32("evenSpread", evenSpread))`

Reply:
Ok. Sounds good. Will add. Most of these log lines may be cleaned out after testing concludes.